File: sources/opencv-samples/1_ReadImage/main.cpp
// https://docs.opencv.org/4.x/db/df5/tutorial_linux_gcc_cmake.html

#include <stdio.h>
#include <opencv2/opencv.hpp>
using namespace cv;

int main(int argc, char** argv)
{
    if ( argc != 2 )
    {
        printf("Usage: ./main.out <Image_Path>\n");
        printf("Read the README.md\n");
        return -1;
    }
    Mat image;
    image = imread( argv[1], IMREAD_UNCHANGED );
    if ( !image.data )
    {
        printf("Could not read the image\n");
        return -1;
    }
    namedWindow("Read Image", WINDOW_AUTOSIZE );
    imshow("Read Image", image);
    std::cout << "Press a key to quit" << std::endl;
    waitKey(0);
    return 0;
}

File: sources/opencv-samples/1_ReadImage/main.py
# https://docs.opencv.org/4.x/db/deb/tutorial_display_image.html

import cv2 as cv
import sys
import argparse


def main():

    args = parser.parse_args()

    path = cv.samples.findFile(args.img_path)
    if not path:
        sys.exit(f"Image {args.img_path} not found")
    img = cv.imread(path, cv.IMREAD_UNCHANGED)

    if img is None:
        sys.exit("Could not read the image")

    cv.imshow("Read Image", img)
    print("Press a key to quit")
    cv.waitKey(0)
    cv.destroyAllWindows();


if __name__ == '__main__':
    
    parser = argparse.ArgumentParser(description="Read Image main.py CLI")
    parser.add_argument("img_path", type=str, help="image path to open")
    main()
